### ISSUE TITLE
Document security implications of using rootless containers

### ISSUE THREAD

**Issue by rutsky** [opencontainers/runc#1659]
"Docker containers are, by default, quite secure" ([cite](https://docs.docker.com/engine/security/security/#conclusions)) - `runc` containers (at least how they are configured in Docker) try to be quite secure for running untrusted code.
It is possible to drop all capabilities, run process inside container as non-root user, etc, and have good enough sandbox for untrusted code (of course with lots of nuances, attack vectors and not found/disclosed vulnerabilities of the Linux kernel).

What are security implications of rootless containers?
Is processes in rootless container are well confined to run untrusted code (in ideal world when there is no vulnerabilities in the Linux kernel)?
How processes from rootless container can influence host system or user processes outside container?

For example, I see that cgroups are not always configured for rootless container, this probably means that process inside rootless container can start fork-bomb and exhaust host resources (just an example, not sure is this true).

It would be nice to have this documented.

**Comment by cyphar** [comment]
Yeah, this is also something that we (or rather I) should document properly in https://rootlesscontaine.rs/.  The general hand-wavey answer I usually give to this question is:

> In principle (assuming there aren't some really awful security vulnerabilities in the kernel that aren't protected against by seccomp/SELinux/AppArmor/etc), a rootless container is **at least** as secure as an unprivileged process on the host -- because we only ever have the privileges we inherited from the unprivileged user. This is something that we get for free that you don't get from a runtime that starts as a root user. However, it also uses the same key isolation features as "rootfull" containers -- so the process is far more restricted that an unprivileged user. The key difference for rootless containers is that you don't need to worry about access control for your runtime daemon that spawns processes as root -- the process that spawns containers can't do anything more than that user that called it.

For your particular question:

> For example, I see that cgroups are not always configured for rootless container, this probably means that process inside rootless container can start fork-bomb and exhaust host resources (just an example, not sure is this true).

The limitations of a rootless container are always restricted by (at least) the user process that started it. So if you're running on a system which sets `pids.max` for all user sessions (like systemd) then a rootless container cannot forkbomb the system any more than 
Assuming there are no critical kernel bugs (that are not protected against by tools like seccomp, AppArmor or SELinux) then a rootless container is at least as secure as an unprivileged process on the host.
